controllers/subscriptions.go

```diff
@@ -42,39 +42,99 @@ import (
 // because fake.Client does not support them.
 func CheckExistingSubscriptions(cli client.Client, desiredSubscription *operatorv1alpha1.Subscription) (*operatorv1alpha1.Subscription, error) {
 
+	odfSub, err := GetOdfSubscription(cli)
+	if err != nil {
+		return nil, err
+	}
+
+	if odfSub.Spec.Config == nil {
+		odfSub.Spec.Config = &operatorv1alpha1.SubscriptionConfig{}
+	}
+
 	subsList := &operatorv1alpha1.SubscriptionList{}
-	err := cli.List(context.TODO(), subsList, &client.ListOptions{Namespace: desiredSubscription.Namespace})
+	err = cli.List(context.TODO(), subsList, &client.ListOptions{Namespace: desiredSubscription.Namespace})
 	if err != nil {
 		return nil, err
 	}
 
+	var subExsist bool
 	var actualSub *operatorv1alpha1.Subscription
 	pkgName := desiredSubscription.Spec.Package
 	for i, sub := range subsList.Items {
 		if sub.Spec.Package == pkgName {
+			subExsist = true
 			if actualSub != nil {
 				foundSubs := []string{actualSub.Name, sub.Name}
 				return nil, fmt.Errorf("multiple Subscriptions found for package '%s': %v", pkgName, foundSubs)
 			}
 			actualSub = &subsList.Items[i]
 			actualSub.Spec.Channel = desiredSubscription.Spec.Channel
 
-			// If the config is not set, only then set it to the desired value => allow user to override
-			if actualSub.Spec.Config == nil {
+			if actualSub.Spec.Config == nil && desiredSubscription.Spec.Config == nil {
```

Comment:
Why not just update https://github.com/red-hat-storage/odf-operator/blob/1969776bdf4ad5bc194cdca0a953fa473e3b6d82/controllers/subscriptions.go#L260 to inject tolerations instead of all these checks?

Add tolerations to desired subscription, and during reconcile merge actual and desired subs. Wouldn't that be simpler?

Reply:
Actually that will require extensive changes in the code, we will have to change the signature of multiple funcs.